Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -252,15 +252,18 @@ RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier id
     index += oidSize + 1; // Read named curve OID, BIT STRING
     if (keyData.size() < index + 1)
         return nullptr;
-    index += bytesUsedToEncodedLength(keyData[index]) + 1; // Read length, InitialOctet
-
-    if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size() - index))
-        return nullptr;
-
+    index += bytesUsedToEncodedLength(keyData[index])+1; // Read length
     CCECCryptorRef ccPublicKey = nullptr;
-    if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data() + index, keyData.size() - index, ccECKeyPublic, &ccPublicKey))
-        return nullptr;
-
+    if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size() - index)) {
+        if (CCECCryptorImportKey(kCCImportKeyCompact, keyData.data() + index+1, keyData.size() - index-1, ccECKeyPublic, &ccPublicKey)) {
```

Comment:
Can you investigate why we need the + 1 here.
Ideally the commit message would indicate why we need it and refer to a spec.
I would tend to write it as:
```
++index;
if (CCECCryptorImportKey(kCCImportKeyCompact, keyData.data() + index, keyData.size() .- index, ...))
```

Reply:
The compressed point format includes the least significant bit of the second coordinates in the first byte to know the orientation , so I think that is why we need one more. I have been doing some debugging and found out that otherwise the function `ccec_compact_import_pub_size` the wrong  number, expecting always a number greater in just one unit.